### PR TITLE
View all competitions

### PR DESCRIPTION
**Pull request by dssherlock**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs [hunk 1]
```diff
@@ -8,13 +8,11 @@ public class DashboardCardBaseModel<T>
     public DashboardCardBaseModel(
         string internalOrgId,
         ICollection<T> items,
-        int numberOfItems,
         bool isDashboardView = false,
         PageOptions pageOptions = null)
     {
         InternalOrgId = internalOrgId;
         Items = items;
-        NumberOfItems = numberOfItems;
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs:17]: `NumberOfItems` was introduced because the Order page presents the total number of orders in the database. With this change, it'll now only show the number of orders based on the search pattern

Comment (dssherlock) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs:17]: Ah i couldnt understand where this had come from! Ok I'll remove this change!

Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs:17]: Discussed this on Teams. We should be able to use `PageOptions` to get the desired behaviour

```csharp
public int NumberOfItems => PageOptions?.TotalNumberOfItems ?? Items.Count
```

`PageOptions` will be null on the dashboard view, so `Items.Count` will be used. Conversely, on the detailed pages, `PageOptions` will not be null and the `TotalNumberOfItems` will then achieve the same result as `OrderIds`

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs [hunk 2]
```diff
@@ -25,9 +23,13 @@ public class DashboardCardBaseModel<T>
 
     public ICollection<T> Items { get; set; }
 
-    public int NumberOfItems { get; set; }
+    public int NumberOfItems => PageOptions?.TotalNumberOfItems ?? Items.Count;
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs:26]: Don't forget the unit tests for this edge-case 😉 

As I'm off next week, feel free to dismiss my feedback in GitHub to not block the PR. @martinkimmings  should also have permission to force-merge if it becomes an issue.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs [hunk 3]
```diff
@@ -25,9 +23,13 @@ public class DashboardCardBaseModel<T>
...
 
     public bool ShouldUsePagination => PageOptions is not null;
 
     public bool IsDashboardView { get; set; }
+
+    public bool HasItems => NumberOfItems > 0;
+
+    public int ItemsToIterate => IsDashboardView ? 5 : NumberOfItems;
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Models/Shared/Dashboard/DashboardCardBaseModel.cs:34]: This'll need a unit test also to validate the logic

**Review by imranbashir01 — APPROVED** [review]

**Review by martinkimmings — APPROVED** [review]
